test-suite/shared/src/test/require-jdk8/org/scalajs/testsuite/javalib/time/YearTest.scala

```diff
@@ -0,0 +1,329 @@
+    assertEquals(10L, firstACYear.until(Year.of(1001), CENTURIES))
+    assertEquals(1L, lastBCYear.until(Year.of(1000), MILLENNIA))
+    assertEquals(1L, firstACYear.until(Year.of(1001), MILLENNIA))
+
+    assertEquals(1L, lastBCYear.until(firstACYear, ERAS))
+    assertEquals(0L, firstACYear.until(Year.of(100), ERAS))
+    assertEquals(0L, lastBCYear.until(Year.of(-100), ERAS))
+  }
+
+  @Test def atDay(): Unit = {
+    for (t <- samples) {
+      assertEquals(LocalDate.of(t.getValue, 1, 1), t.atDay(1))
+      assertEquals(LocalDate.of(t.getValue, 12, 31), t.atDay(t.length()))
+    }
+
+    assertEquals(LocalDate.of(leapYear.getValue, 2, 29), leapYear.atDay(31 + 29))
+    assertEquals(LocalDate.of(leapYear.getValue, 3, 1), leapYear.atDay(31 + 30))
+
+    assertEquals(LocalDate.of(nonLeapYear.getValue, 3, 1), nonLeapYear.atDay(31 + 29))
+  }
+
+  @Test def isLeap(): Unit = {
+    assertTrue(leapYear.isLeap)
+    assertFalse(nonLeapYear.isLeap)
+
+    assertTrue(Year.isLeap(leapYear.getValue))
+    assertFalse(Year.isLeap(nonLeapYear.getValue))
+  }
+
+  @Test def isValidMonthDay(): Unit = {
+    val leapMonthDay = MonthDay.of(2, 29)
+    assertTrue(leapYear.isValidMonthDay(leapMonthDay))
+    assertFalse(nonLeapYear.isValidMonthDay(leapMonthDay))
+
+    val otherMonthDays = Month.values().map(m => MonthDay.of(m, m.minLength()))
+    for {
+      t <- samples
+      monthDay <- otherMonthDays
+    } {
+      assertTrue(t.isValidMonthDay(monthDay))
+    }
+  }
+
+  @Test def atMonthDay(): Unit = {
+    val leapMonthDay = MonthDay.of(2, 29)
+    val leapDate = LocalDate.of(leapYear.getValue(),
+        leapMonthDay.getMonthValue, leapMonthDay.getDayOfMonth)
+    assertEquals(leapDate, leapYear.atMonthDay(leapMonthDay))
+
+    val nonLeapDate = LocalDate.of(nonLeapYear.getValue(),
+      leapMonthDay.getMonthValue, 28)
+    assertEquals(nonLeapDate, nonLeapYear.atMonthDay(leapMonthDay))
+
+    val otherMonthDays = Month.values().map(m => MonthDay.of(m, m.minLength()))
+    for {
+      t <- samples
+      monthDay <- otherMonthDays
+    } {
+      val expected = LocalDate.of(t.getValue, monthDay.getMonthValue, monthDay.getDayOfMonth)
+      assertEquals(expected, t.atMonthDay(monthDay))
+    }
+  }
+
+  @Test def compareTo(): Unit = {
+    assertTrue(min.compareTo(min) == 0)
+    assertTrue(min.compareTo(max) < 0)
+    assertTrue(max.compareTo(min) > 0)
+    assertTrue(max.compareTo(max) == 0)
+  }
+
+  @Test def isAfter(): Unit = {
+    assertFalse(min.isAfter(min))
+    assertFalse(min.isAfter(max))
+    assertTrue(max.isAfter(min))
+    assertFalse(max.isAfter(max))
+  }
+
+  @Test def isBefore(): Unit = {
+    assertFalse(min.isBefore(min))
+    assertTrue(min.isBefore(max))
+    assertFalse(max.isBefore(min))
+    assertFalse(max.isBefore(max))
+  }
+
+  @Test def now(): Unit = {
+    val now = LocalDate.now()
+    val year = Year.now()
+
+    assertEquals(now.getYear, year.getValue)
```

Comment:
This could fail if executed exactly on at new year. I would suggest to simply add a small message to the `assertEquals`: `assertEquals("Test for now failed on the current year (this test could fail if executed at new year).", now.getYear, year.getValue)`  or:

```scala
val now = LocalDate.now()
val year = Year.now()
if (now.getYear != year.getValue) {
  assertEquals(LocalDate.now(), year)
  println("Happy new year " + year + '!')
}

```